### ISSUE TITLE
Authentication prompt can be bypassed on some devices via back button

### ISSUE THREAD

**Issue by gmale** [zcash/zcash-android-wallet#215]
While sending funds, on devices that show an authentication prompt (rather than facial recognition), pressing back during that prompt continues on to the sending screen.

Steps:
1.  create transaction
2.  press send
3.  observe authentication prompt (rather than facial recognition -- this is device and config dependent)
4.  press back
5.  observe: app continues to sending screen
6.  expect: app returns to transaction creation screen

Needs investigation but this is likely an issue with trying to ignore failures gracefully on certain devices while respecting them on other devices.

**Comment by gmale** [comment]
Note: Geffen's nexus 5 and HTC had a related issue where send would not work when no passcode was set on the device. We need to find a happy middle-ground behavior. So once this is fixed for the given device, try it again on the test phones!

**Comment by gmale** [comment]
There was an issue in the Android biometrics library where `ERROR_USER_CANCELED` was being returned in certain cases where devices did not have a screen lock configured. This lead me to allowing those devices to bypass authorization. That has been fixed in the underlying library and I've made improvements to handle all use cases better.

I've tested on an HTC A9 running Android 7, a Nexus 5 running Android 5.1.1 and a Pixel 4 XL on Android 11, with and without faceid/PINs/Passwords/Patterns set.

Unfortunately I don't have funds readily available to test the send flow so I haven't yet merged this but I'm pretty confident it now works in this scenario.
